scrapinghub/client/collections.py

```diff
@@ -125,11 +125,18 @@ class Collection(object):
         >>> foo_store.iter()
         <generator object jldecode at 0x1049eef10>
 
-    - iterate iterate over _key & value pair::
+    - iterate over _key & value pair::
 
         >>> for elem in foo_store.iter(count=1)):
         ...     print(elem)
         [{'_key': '002d050ee3ff6192dcbecc4e4b4457d7', 'value': '1447221694537'}]
+        
+    - get 1000th item key::
+    
+        >>> import itertools
+        >>> keys = foo_store.iter(nodata=True, meta=["_key"]))
+        >>> next(itertools.islice(keys, 1000, 1001))
```

Comment:
If your list has only 1k elements (and it's important because iter & list handlers return up to 1k elements max by default), this won't work:
```
>>> next(itertools.islice([1,2,3], 3, 4))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
StopIteration
```
Also, getting Nth item's key is simpler (and more lightweight) by providing index param or getting item by the index, isn't it? In this case, you don't need to go through elements at all. There might be additional filters, so it probably won't work as you want in all cases, but as a general approach for an abstract scenario (i.e. as an example in the docs) that's a way better.
```
>>> job.items.list(index=1000)
{'field': 'data', ...}

>>> job.items.get(1000)
{'field': 'data', ...}
```
I'd suggest you to pick another example to demonstrate the nodata feature, smth like:
```
- iterate over keys only (doesn't fetch items data)

    >>> for elem in foo_store.iter(nodata=True, meta=["_key"])):
    ...     print(elem)
    {'_key': '002d050ee3ff6192dcbecc4e4b4457d7'}
```

Reply:
I thought it's obvious that we know our dataset size is bigger then 1000, isn't it?

> (and it's important because iter & list handlers return up to 1k elements max by default)
It seems they return all elements by default, and islice is the fastest way https://jupyter.scrapinghub.com/user/v/lab/tree/shared/Experiments/Arche/collections_nodata.ipynb

Reply:
> I thought it's obvious that we know our dataset size is bigger then 1000, isn't it?

Maybe, but  it's very common that users are glancing through the examples very fluently, and their assumptions can be totally different :slightly_smiling_face: Thanks!